spec/requests/assessments_spec.rb

```diff
@@ -0,0 +1,91 @@
+require 'rails_helper'
+
+RSpec.describe "Assessments", type: :request do
+  describe "POST /assessments/assess" do
+    subject(:dispatch_request) { post "/assessments/assess", params:, headers: }
+    let(:params) { { account_id: user.account_id } }
+    let(:user) { create(:user) }
+    let(:headers) do
+      {
+        "ACCEPT" => "application/json",
+        SignupsController::INCOGNIA_INSTALLATION_ID_HEADER => installation_id
+      }
+    end
+    let(:installation_id) { SecureRandom.hex }
+
+    before do
+      allow(Assessments::AssessForm).to receive(:new).and_return(form)
+    end
+    let(:form) do
+      instance_double(
+        Assessments::AssessForm,
+        errors: form_errors,
+        submit: submit_return
+      )
+    end
+    let(:form_errors) { [] }
+    let(:submit_return) { nil }
+
+    context 'when validations succeed' do
+      let(:submit_return) { assessments }
+      let(:assessments) { build_list(:assessments_assessment, 2) }
+
+      it "invokes assess form" do
+        expect(Assessments::AssessForm).to receive(:new)
+          .with(user:, installation_id:)
+          .and_return(form)
+
+        expect(form).to receive(:submit)
+
+        dispatch_request
+      end
+
+
+      it "returns http success" do
+        dispatch_request
+
+        expect(response).to have_http_status(:success)
+      end
+
+      it "returns requested assessments as JSON" do
+        dispatch_request
+
+        serializable_assessments = assessments.map do |assessment|
+          AssessmentSerializer.new(assessment:)
+        end
+
+        expect(response.body).to eq(serializable_assessments.to_json)
+      end
+
+      it_behaves_like 'handle Incognia API errors' do
+        let(:service) { form }
+        let(:method) { :submit }
+      end
+    end
+
+    context 'when validations fails' do
```

Comment:
What happens when can't find the user?

https://github.com/inloco/incognia-demo-api-ruby/pull/29/files#diff-cf4c5fba34af4eeec826dad9c4c9380440487102292e3f91bf40801086e06ffaR4

Reply:
Holly shit. I thought it was implemented to every child of `ApplicationController`.

I'm gonna open a new PR with some error-handling adjustments. :praise:

Reply:
https://github.com/inloco/incognia-demo-api-ruby/pull/31/files